pkg/controllers/data.go

```diff
@@ -179,3 +179,19 @@ func getQueryParams(r *http.Request) (query *entities.Query, err error) {
 		FinishDate: finishDate,
 	}, nil
 }
+func (d *DataController) writeResponseByError(w http.ResponseWriter, err error) {
+	switch err.Error() {
+	case "user is not authorized to list thing's data":
+		d.writeResponse(w, http.StatusBadRequest, err.Error())
+	case "no access token provided":
+		d.writeResponse(w, http.StatusBadRequest, err.Error())
```

Comment:
Should missing field errors result in a bad request error? Could you please mention the references you have used?

Reply:
I found this [discussion](https://stackoverflow.com/questions/16133923/400-vs-422-response-to-post-of-data) on StackOverflow and it seems good.

Reply:
I saw that BadRequest can be used for missing data [here](https://www.restapitutorial.com/httpstatuscodes.html). Anyway, I'm going to read the discussion that you mentioned.

Reply:
Ack.